deploy/runtools/runtime_config.py

```diff
@@ -97,6 +106,20 @@ def __init__(self, name: str, hwconfig_dict: Dict[str, Any]) -> None:
         self.tarball_built = False
 
         self.additional_required_files = []
+    
+    def __validate_uri(self, hwconfig_dict: Dict[str, Any], prop: str) -> Optional[str]:
+        """ Private method to init / validate any URI. """
+
+        uri = hwconfig_dict.get(prop)
+
+        if uri is None:
+            return None
+
+        is_uri = re.match(_RFC_3986_PATTERN, uri)
+        if not is_uri:
+            raise Exception(f"when {prop} is set, it must be a URI. (found '{uri}')")
```

Comment:
In https://github.com/firesim/firesim/pull/1140/files#r927042552 @davidbiancolin wanted to support things optionally being a filepath because URIs don't have a means for specifying a relative path.  His thinking may have changed on that though.

Reply:
I spoke to @sifive-benjamin-morse about this out-of-band (after you made this comment) and i really don't have a strong opinion anymore (so i'm deferring to Ben). I guess there's a strong possibility i'll want the feature when i start using the mangaer.

Whether non-URI, relative paths should be supported is probably a question best posed to the `firesim-dev` chat.